skietbaan_fe/src/components/Login.js

```diff
@@ -1,98 +1,135 @@
+      validForm:false
     }
-    this.showAlert = this.showAlert.bind(this);
+    this.Login = this.Login.bind(this);
     this.handleChange = this.handleChange.bind(this);
-    this.postData = this.handleChange.bind(this);
   }
   handleChange({ target }) {
     this.setState({
-      [target.name]: target.value
+      [target.name]: target.value,
+      invalidPassword:false,
+      invalidEmail: false,
+      invalidUsername: false
     });
-  }
-   postData(url = ``, data = {}) {
-    // Default options are marked with *
-      return fetch(url, {
-          method: "POST", // *GET, POST, PUT, DELETE, etc.
-          mode: "cors", // no-cors, cors, *same-origin
-          cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
-          credentials: "same-origin", // include, *same-origin, omit
-          headers: {
-              "Content-Type": "application/json",
-              // "Content-Type": "application/x-www-form-urlencoded",
-          },
-          redirect: "follow", // manual, *follow, error
-          referrer: "no-referrer", // no-referrer, *client
-          body: JSON.stringify(data), // body data type must match "Content-Type" header
-      })
-      .then(response => response.json()); // parses response to JSON
-  }
-  
-  showAlert() {
-
 
-    if(this.state.passwordValue === "" || this.state.passwordValue === null)
+    var Valid = true;
+    if(!validatePassword(this.state.passwordValue))
+    {
+        this.setState({
+          invalidPassword : true
+        });
+        Valid = false;
+    }
+    if(!validateEmail(this.state.emailValue))
+    {
+        this.setState({
+          invalidEmail : true
+        });
+        Valid = false;
+    }
+    if(validateUsername(this.state.usernameValue))
+    {
+        this.setState({
+          invalidUsername : true
+        });
+        Valid = false;
+    }
     this.setState({
-      invalid : true
+      validForm:Valid
     });
-
-
+  }
+  
+  Login() {
+    if(this.state.validForm)
+    {
     var RequestObject = {
-        "Username":this.state.usernameValue,
-        "Password":this.state.emailValue,
-        "Surname":this.state.passwordValue
+      "Username":this.state.usernameValue,
+      "Password":this.state.emailValue,
+      "Surname":this.state.passwordValue
     }
     console.log(RequestObject);
 
-
-     fetch('http://skietbaan.retrotest.co.za/api/User', {
-     method: 'post',
+    fetch('http://skietbaan.retrotest.co.za/api/values', {
+      method: 'post',
       headers: {
         'Accept': 'application/json',
         'Content-Type': 'application/json'
-      },
-     body: JSON.stringify(RequestObject)
+       },
+       body: JSON.stringify(RequestObject)
       }).then(function(response) {
-        console.log('Created Gist:');
         return response.json();
       }).then(function(data) {
-        console.log('Created Gist:', data.html_url);
+        console.log('Backend output', data);
+      }).catch(function() {
```

Comment:
You can replace 
```
function() {
  //some stuff
}
```
with
```
( ) => {
  //some stuff
}
```
It's the es6 shorthand for declaring functions.

Reply:
alright changed

Reply:
You might want to keep it as a function since ie11 doesnt support this and a ton of people still use ie11.

https://stackoverflow.com/questions/40216015/arrow-function-not-working-in-ie-why
https://caniuse.com/#feat=arrow-functions

Reply:
ok will change back

Reply:
Lol. Stupid IE. *rolling-eyes*